### DISCUSSION TITLE
The AutoGPT architecture requires loops, not an iteration based on arrays.

### DISCUSSION THREAD

**Discussion by pengyuxiang1** [langgenius/dify#5891] [Suggestion]
As the title suggests, I am aiming to implement a flowchart for the autogpt that allows for the orchestration of looped functionalities such as "thinking" and "execution," but currently, the "iteration" component does not support this.  This cyclic structure is crucial, and it's hoped that support can be added.

**Comment by zhangxin840** [comment]
Looping is needed to support the basic workflow of the review-refine process, which is also essential for multi-agent collaboration.

**Comment by sevenjay** [comment]
someone support this feature
https://github.com/langgenius/dify/issues/12354
